src/NzbDrone.Core/Notifications/MediaBrowser/MediaBrowserSettings.cs

```diff
@@ -12,38 +13,37 @@ public MediaBrowserSettingsValidator()
         {
             RuleFor(c => c.Host).ValidHost();
             RuleFor(c => c.ApiKey).NotEmpty();
+            RuleFor(c => c.UrlBase).ActuallyValidUrlBase();
         }
     }
 
     public class MediaBrowserSettings : IProviderConfig
     {
         private static readonly MediaBrowserSettingsValidator Validator = new MediaBrowserSettingsValidator();
 
-        public MediaBrowserSettings()
-        {
-            Port = 8096;
-        }
-
         [FieldDefinition(0, Label = "Host")]
         public string Host { get; set; }
 
         [FieldDefinition(1, Label = "Port")]
-        public int Port { get; set; }
+        public int Port { get; set; } = 8096;
 
         [FieldDefinition(2, Label = "Use SSL", Type = FieldType.Checkbox, HelpText = "Connect to Emby over HTTPS instead of HTTP")]
         public bool UseSsl { get; set; }
 
-        [FieldDefinition(3, Label = "API Key", Privacy = PrivacyLevel.ApiKey)]
+        [FieldDefinition(3, Label = "Url Base", Type = FieldType.Textbox, Advanced = true, HelpText = "Adds a prefix to the Emby server url, see http://[host]:[port]/[urlBase]")]
+        public string UrlBase { get; set; } = "";
+
+        [FieldDefinition(4, Label = "API Key", Privacy = PrivacyLevel.ApiKey)]
         public string ApiKey { get; set; }
 
-        [FieldDefinition(4, Label = "Send Notifications", HelpText = "Have MediaBrowser send notfications to configured providers", Type = FieldType.Checkbox)]
+        [FieldDefinition(5, Label = "Send Notifications", HelpText = "Have MediaBrowser send notfications to configured providers", Type = FieldType.Checkbox)]
         public bool Notify { get; set; }
 
-        [FieldDefinition(5, Label = "Update Library", HelpText = "Update Library on Import & Rename?", Type = FieldType.Checkbox)]
+        [FieldDefinition(6, Label = "Update Library", HelpText = "Update Library on Import & Rename?", Type = FieldType.Checkbox)]
         public bool UpdateLibrary { get; set; }
 
         [JsonIgnore]
-        public string Address => $"{Host}:{Port}";
+        public string Address => $"{(UseSsl ? "https" : "http")}://{Host}:{Port}{(String.IsNullOrEmpty(UrlBase) ? "" : $"/{UrlBase}")}";
```

Comment:
This should be built in the proxy. Instead of passing individual values to those proxy functions the settings object can be passed in, follow what is already being done in SabnzbdProxy: https://github.com/Sonarr/Sonarr/blob/7f079ac8f29ae5972e85fc53c4951c1882329595/src/NzbDrone.Core/Download/Clients/Sabnzbd/SabnzbdProxy.cs#L38

Same applies for the other media servers.

Reply:
Reason I have it in the 'Settings' is that it can be referenced infinite times. For example Xmbc uses it 16 times, which would require me to write the following line 16 times into the proxy.

Can I keep the Address in the Settings, but use either the HttpRequestBuilder or another common utility, to format the string?

Reply:
No, I don't think this belongs in settings. XBMC was already using `HttpRequestBuilder.BuildBaseUrl` in `SendCommand` / `ProcessRequest`. Why would it have to be written 16 times in the proxy?

Reply:
There are 16 references to `XbmcSettings.Address` in 13 functions, where the settings come as parameters. This would mean, I have to write `var baseUrl = HttpRequestBuilder.BuildBaseUrl(settings.UseSsl, settings.Host, settings.Port, settings.UrlBase);` 13 times at those locations.

`Address` is a "hidden" member of many settings, including download clients, connections. If you want to change settings to include only pure members, then it will either result in code duplication or requires structural changes for proxies to be able cacheing these internally.

Reply:
Then I will remove the `Address` property from the settings of these connections. ( It will cause some mismatch with others as they have this `Address` field, but I rather not refactor everything in a single pr. )

Reply:
Yeah, that's reasonable, more so-not make it worse and we can clean up the remaining afterwards.